jsonschema-core.xml

```diff
@@ -800,49 +800,302 @@
+                to impose additional constraints such as forbidding certain keywords, or
+                performing unusually strict syntactical validation, as might be done
+                during a development and testing cycle.
             </t>
+            <section title='The "$schema" Keyword'>
+                <t>
+                    The "$schema" keyword is both used as a JSON Schema feature set identifier and
+                    the location of a resource which is itself a JSON Schema, which describes any
+                    schema written for this particular feature set.
+                </t>
+                <t>
+                    The value of this keyword MUST be a <xref target="RFC3986">URI</xref>
+                    (containing a scheme) and this URI MUST be normalized.
+                    The current schema MUST be valid against the meta-schema identified by this URI.
+                </t>
+                <t>
+                    If this URI identifies a retrievable resource, that resource SHOULD be of
+                    media type "application/schema+json".
+                </t>
+                <t>
+                    The "$schema" keyword SHOULD be used in a root schema.
+                    It MUST NOT appear in subschemas.
+                </t>
+                <t>
+                    <cref>
+                        Using multiple "$schema" keywords in the same document would imply that the
+                        feature set and therefore behavior can change within a document.  This would
+                        necessitate resolving a number of implementation concerns that have not yet
+                        been clearly defined.  So, while the pattern of using "$schema" only in root
+                        schemas is likely to remain the best practice for schema authoring,
+                        implementation behavior is subject to be revised or liberalized in
+                        future drafts.
+                    </cref>
+                    <!--
+                        In particular, the process of validating an instance, including validating a
+                        schema as an instance against its meta-schema, only allows for a single set
+                        of rules across the entire instance document.  There is no equivalent of
+                        changing the meta-schema partway through the validation for non-schema
+                        instances.
+                    -->
+                </t>
+                <t>
+                    Values for this property are defined elsewhere in this and other documents,
+                    and by other parties.
+                </t>
+            </section>
+            <section title='The "$vocabulary" Keyword'>
+                <t>
+                    The "$vocabulary" keyword, which appears in a meta-schema, declares
+                    what sets of keywords are expected to be used in schemas described
```

Comment:
Call me picky, but the text "declares what sets of keywords" doesn't make me associate that the property names in `$vocabularies` are identifiers for vocabularies, it may make the reader expect some list of keywords or something. In this section the word "identifier" does not appear at all (just "identifying" in a cref). In contrast, in the "$schema" section it is clearly indicated that `$schema` is a "feature set identifier".

To reinforce the fact that the names of the properties in this keyword are identifiers for vocabularies, something in the likes of "The names of the object properties are used as JSON Schema vocabulary identifiers." might be stated somewhere (maybe in next paragraph after indicating that the keyword must be an object, or on a paragraph on its own just after that).

Reply:
I like the new wording, much clearer! :smile: